src/music_gui.cpp

```diff
@@ -834,6 +828,38 @@ struct MusicWindow : public Window {
 				break;
 		}
 	}
+
+	virtual void OnMouseDrag(Point pt, int widget)
+	{
+		this->UpdateVolumeWidget(pt);
+	}
+
+	void UpdateVolumeWidget(Point pt)
+	{
+		uint widget_width = this->GetWidget<NWidgetBase>(this->dragged_vol_widget)->current_x;
+
+		int x = pt.x - this->GetWidget<NWidgetBase>(this->dragged_vol_widget)->pos_x;
+
+		/* Only update the volume along the width of this widget */
+		if (x < 0) x = 0;
+		if (x > (int)widget_width) x = (int)widget_width;
+
+		byte *vol = (this->dragged_vol_widget == WID_M_MUSIC_VOL) ? &_settings_client.music.music_vol : &_settings_client.music.effect_vol;
+
+		byte new_vol = x * 127 / widget_width;
+		if (_current_text_dir == TD_RTL) new_vol = 127 - new_vol;
+		/* Clamp to make sure min and max are properly settable */
+		if (new_vol > 124) new_vol = 127;
+		if (new_vol < 3) new_vol = 0;
+		if (new_vol != *vol) {
+			/* Update the volume in the game's settings */
+			*vol = new_vol;
+			/* If this slider is for the music volume, we also need to
+			   update the music driver's volume. */
```

Comment:
Ooh, nit picking but this comment should either be one line or start with an asterisk.